python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,1093 @@
+        self._mask = MultibandPixel(filters, mask, bbox=self.getBBox())
+        self._variance = MultibandPixel(filters, variance, bbox=self.getBBox())
+
+    def copy(self, deep=False):
+        """Make a copy of the current instance
+
+        `MultibandPixel._singles` is an array,
+        so this just makes a copy of the array
+        (as opposed to a view of the parent array).
+        """
+        if deep:
+            filters = tuple([f for f in self.filters])
+            singles = tuple([s.copy(True) for s in self.singles])
+            # For MultibandPixels, `bbox` is a `Point2I`
+            coords = self.getBBox()
+            bbox = Point2I(coords.getX(), coords.getY())
+        else:
+            filters = self.filters
+            singles = self.singles
+            bbox = self.getBBox()
+        return MultibandMaskedPixel(filters=filters, singles=singles, bbox=bbox)
+
+    def __getitem__(self, indices):
+        """Get a slice of the underlying array
+
+        Since a `MultibandPixel` is a scalar in the
+        spatial dimensions, it can only be indexed with
+        a filter name, number, or slice.
+        """
+        image = self.image[indices]
+        mask = self.mask[indices]
+        variance = self.variance[indices]
+
+        if hasattr(image, "__len__"):
+            result = MultibandMaskedPixel(filters=self.filters, image=image, mask=mask,
+                                          variance=variance, bbox=self.getBBox())
+        else:
+            result = MaskedPixel(image, mask, variance, self.getBBox())
+        return result
+
+    def setBBox(self, bbox):
+        """Overload `MultibandBase.setBBox`
+
+        Since the "single band objects" are elements
+        of an array and do not have a bounding box,
+        overloading `setBBox` is needed to prevent the
+        code from attempting to set the bounding box of
+        each item in `singles`.
+
+        Parameters
+        ----------
+        bbox: Point2I
+            Pixel location in the parent image
+        """
+        if not isinstance(bbox, Point2I):
+            err = "The bounding box for a `MultibandPixel` should always be a `Point2I`, received {0}"
+            raise ValueError(err.format(bbox))
+        self._bbox = bbox
+
+
+class MultibandMaskedImage(MultibandTripleBase):
+    """MultibandMaskedImage class
+
+    This class acts as a container for multiple `afw.MaskedImage` objects.
+    All masked images must have the same bounding box, and the associated
+    images must all have the same data type.
+    """
+    def __init__(self, filters, singles=None, image=None, mask=None, variance=None,
+                 filterKwargs=None, **kwargs):
+        """Initialize a `MultibandMaskedImage` object
+
+        See `MultibandTripleBase` for parameter definitions.
+        """
+        super().__init__(filters, singles, image, mask, variance, MaskedImageF, filterKwargs, **kwargs)
+        self._bbox = self.singles[0].getBBox()
```

Comment:
It actually wasn't because at the time `MultibandMaskedPixel` also inherited from `MultibandTripleBase`, so `_bbox` was not set in `super`. It is now and I removed this line.